src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java

```diff
@@ -852,4 +857,32 @@ public void cancelOnlyRelevantQueueItem() throws Exception {
         r.assertBuildStatus(Result.ABORTED, r.waitForCompletion(b));
         r.assertLogContains("ran on special agent", b);
     }
+
+    @Test
+    public void garbageCollection() throws Exception {
+        // Pod exists, need to kill the build, delete the agent without deleting the pod.
+        // Wait for the timeout to expire and check that the pod is deleted.
+        var garbageCollection = new GarbageCollection();
+        garbageCollection.setTimeout(120);
+        cloud.setGarbageCollection(garbageCollection);
+        r.jenkins.save();
+        r.waitForMessage("Running on remote agent", b);
+
+        Pod pod = null;
+        for (Computer c : r.jenkins.getComputers()) {
+            if (c instanceof KubernetesComputer) {
+                var node = (KubernetesSlave) c.getNode();
+                pod = node.getPod().get();
+                Assert.assertNotNull(pod);
+                r.jenkins.removeNode(node);
+                break;
+            }
+        }
+        // Build is marked as failed because the agent has vanished
+        r.assertBuildStatus(Result.FAILURE, r.waitForCompletion(b));
```

Comment:
OK I guess but not very realistic.

Maybe you can `doKill` the build to simulate a loss of regular cleanup from `podTemplate`?